modules/dxp/apps/antivirus/antivirus-async-store/src/main/java/com/liferay/antivirus/async/store/internal/AntivirusAsyncDLStore.java

```diff
@@ -411,6 +394,29 @@ private void _registerCallback(DLStoreRequest dlStoreRequest)
 			});
 	}
 
+	private void _validate(
+			String fileName, String fileExtension, String sourceFileName,
+			boolean validateFileExtension, String versionLabel,
+			InputStream inputStream)
+		throws PortalException {
+
+		_dlValidator.validateFileName(fileName);
+
+		if (validateFileExtension) {
+			_dlValidator.validateFileExtension(fileName);
+		}
+
+		_dlValidator.validateSourceFileExtension(fileExtension, sourceFileName);
+
+		_dlValidator.validateVersionLabel(versionLabel);
+
+		if (inputStream != null) {
+			_dlValidator.validateFileSize(
+				GroupThreadLocal.getGroupId(), fileName,
+				MimeTypesUtil.getContentType(fileName), inputStream);
+		}
+	}
+
```

Comment:
This method would raise an exception if parameter `sourceFilename` is not null and `fileExtension` is null, when calling `validateSourceFileExtension`. Maybe a fix is needed in `validateSourceFileExtension`?